net/instaweb/rewriter/css_inline_filter.cc

```diff
@@ -111,11 +112,22 @@ void CssInlineFilter::StartDocumentImpl() {
 
 CssInlineFilter::~CssInlineFilter() {}
 
+void CssInlineFilter::StartElementImpl(HtmlElement* element) {
+  if (element->keyword() == HtmlName::kBody) {
+    in_body_ = true;
+  }
+}
+
 void CssInlineFilter::EndElementImpl(HtmlElement* element) {
   // Don't inline if the CSS element is under <noscript>.
   if (noscript_element() != NULL) {
     return;
   }
+
+  if (element->keyword() == HtmlName::kBody) {
+    in_body_ = false;
```

Comment:
In case <body> element comes before <head>, so thought of making in_body_ to false in EndElementImpl(). But actually for w3c compatibility, this original structure itself will not be compatible. So we can remove this.  
I can also rename this flag as you mentioned.